modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/odata/ExpressionVisitorImpl.java

```diff
@@ -230,7 +249,7 @@ else if (type == MethodExpression.Type.NOW) {
 						"type ", type, " and ", expressions.size(), "params"));
 			}
 
-			return String.valueOf(type);
+			return MethodExpression.Type.NOW;
```

Comment:
I am not sure about this, because it couples the Result to the Expression, I think would be better to create your custom type, at this level.

Reply:
But currently the result is totally couples with the expression because we are producing the result from the expression or maybe I am wrong ?

Reply:
We are reading the expression, so the result should no be coupled with the expression, but to be another layer.

Reply:
Got it.  I am going to add new type to avoid that coupling. 

Thanks @cgoncas

Reply:
Awesome!!

Once you do it, forward it :)